homeworks/asztaloslajos/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,121 @@
+"""
+to_do.py --- Asztalos Lajos --- 2024.10.21
+setup_logger function by Nagy István Gábor 
+"""
+import logging
+
+#parameters
+MENU = list(["Feladat hozzáadása","Feladatok listázása","Feladat törlése", "Kilépés"])
+FILEPATH = "homeworks/asztaloslajos/4_exceptions_logging/"
+FILENAME = "tasks.txt"
+TXT_READ_INFO ="The tasks have been read from the file"
+TXT_WRITE_INFO ="The task has been written to the file"
+
+#functions
+#log functions
+def create_file_handler(log_file, level):
+    file_handler = logging.FileHandler(log_file)
+    file_handler.setLevel(level)
+    return file_handler
+
+def create_stream_handler(level):
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(level)
+    return stream_handler
+
+def set_formatter(handler):
+    log_format = logging.Formatter('%(asctime)s - %(name)s - %(module)s - %(levelname)s - %(message)s')
+    handler.setFormatter(log_format)
+
+def setup_logger(name, log_file="homeworks/asztaloslajos/4_exceptions_logging/app.log", level=logging.INFO, handlers=None):
+
+    logger = logging.getLogger(name)
+    logger.setLevel(level)
+
+    if handlers is None:
+        handlers = ['file', 'stream']
+    
+    for handler in handlers:
+        if handler == 'file':
+            file_handler = create_file_handler(log_file, level)
+            set_formatter(file_handler)
+            logger.addHandler(file_handler)
+        elif handler == 'stream':
+            stream_handler = create_stream_handler(level)
+            set_formatter(stream_handler)
+            logger.addHandler(stream_handler)
+    
+    return logger
+
+logger = setup_logger("applog", "homeworks/asztaloslajos/4_exceptions_logging/app.log", level="DEBUG")
+
+#task functions
+def add_task(param):
+    try:
+        with open(FILEPATH + FILENAME, "a") as file:
+            file.write(f"{param}\n")
+    except Exception as e:
+        logger.error(e)
+    else: logger.info(TXT_WRITE_INFO)
+
+def view_task():
+    try:
+        with open(FILEPATH + FILENAME, "r") as file:
+            tasks = file.readlines()
+    except Exception as e:
+        logger.error(e)
+    else: logger.info(TXT_READ_INFO)
+    for id, task in enumerate(tasks, 1):
+        print(f"{id} - {task.strip()}")
+
+def remove_task(param):
```

Comment:
Szuper lenne ha a remove_task ki is listázná a feladatokat az id-val együtt, hogy tudja a felhasználó hogy melyik feladathoz milyen szám tartozik, megkönnyítve a törlést.
Azt is jó lenne megnézni, hogy a felhasználó olyan számot adott e meg ami a listában van. Ha nem, akkor erre egy megfelelő hibaüzenetet lehetne kiírni.